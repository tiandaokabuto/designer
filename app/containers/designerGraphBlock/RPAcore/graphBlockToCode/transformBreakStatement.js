/**
 * (已进行单元测试)
 * @param {*} padding
 * @param {*} dataStructure
 * @param {*} result
 */
const transformBreakStatement = (padding, dataStructure, result) => {
  result.output += `${padding}`;
  result.output += `break`;
  return result.output;
};

export default transformBreakStatement;
