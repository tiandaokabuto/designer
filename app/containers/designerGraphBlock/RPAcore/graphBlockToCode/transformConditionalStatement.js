const transformConditionalStatement = (padding, dataStructure, result) => {
  // const looptype = dataStructure['properties']['required'][0].value;
  // const loopcondition = dataStructure['properties']['required'][1].value;
  // result.output += `${padding}${looptype} ${loopcondition}:\n`;
  const { tag, value, valueList = [] } = dataStructure['properties'][
    'required'
  ][0];
  if (tag === 2) {
    // 自定义
    const loopcondition = dataStructure['properties']['required'][0].value;
    result.output += `${padding}if ${loopcondition}:\n`;
  } else {
    // 向导
    result.output += `${padding}if `;
    valueList.forEach((item, index) => {
      if (index === valueList.length - 1) {
        // 最后一个，不把连接符填上
        if (item.rule === 'is None' || item.rule === 'not None') {
          result.output += `(${item.v1} ${item.rule}) `;
        } else {
          result.output += `(${item.v1} ${item.rule} ${item.v2}) `;
        }
      } else {
        if (item.rule === 'is None' || item.rule === 'not None') {
          result.output += `(${item.v1} ${item.rule}) ${item.connect} `;
        } else {
          result.output += `(${item.v1} ${item.rule} ${item.v2}) ${item.connect} `;
        }
      }
    });
    result.output += `:\n`;
  }
};

export default transformConditionalStatement;
