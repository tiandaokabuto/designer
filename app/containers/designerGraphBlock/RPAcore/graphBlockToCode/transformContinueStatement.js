/**
 * (已进行单元测试)
 * @param {*} padding
 * @param {*} dataStructure
 * @param {*} result
 */
const transformContinueStatement = (padding, dataStructure, result) => {
  result.output += `${padding}`;
  result.output += `continue`;
  return result.output;
};

export default transformContinueStatement;
