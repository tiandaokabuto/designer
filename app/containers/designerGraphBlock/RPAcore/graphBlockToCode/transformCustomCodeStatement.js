import memoize from './reselect';
/**
 * (已进行单元测试)
 * @param {*} padding
 * @param {*} dataStructure
 * @param {*} result
 */
const transformCustomCodeStatement = (padding, dataStructure, result) => {
  if (dataStructure.codeValue) {
    const ignore = dataStructure.ignore ? '# ' : '';
    console.log(dataStructure.codeValue);
    const buf = dataStructure.codeValue
      .split('\n')
      .map(item => padding + ignore + item)
      .join('\n');
    // result.output += `${padding}${ignore}\n`;
    result.output += buf;
    // result.output += `\n${padding}${ignore}`;
  }
  return result.output;
};

export default memoize(transformCustomCodeStatement);
