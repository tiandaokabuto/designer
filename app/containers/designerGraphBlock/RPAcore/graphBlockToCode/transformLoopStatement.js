const transformLoopStatement = (padding, dataStructure, result) => {
  console.log(dataStructure, 'dataStructure');
  const select = dataStructure['properties']['required'][0].value;
  const node = dataStructure['properties']['required'][1];
  const valueConditionList = node.valueList;
  let loopcondition = '';
  const looptype = select === 'for_condition' ? 'while' : 'for';
  if (select === 'for_list' && node[select]) {
    loopcondition = `${node[select][0].value} in ${node[select][1].value}`;
  } else if (select === 'for_dict' && node[select]) {
    loopcondition = `${node[select][0].value},${node[select][1].value} in ${node[select][2].value}.items()`;
  } else if (select === 'for_times' && node[select]) {
    loopcondition = `${node[select][0].value} in range(${node[select][1].value},${node[select][2].value},${node[select][3].value})`;
  } else if (select === 'for_condition' && node.tag === 2) {
    loopcondition = node.value;
  } else if (
    select === 'for_condition' &&
    node.tag === 1 &&
    Array.isArray(valueConditionList)
  ) {
    loopcondition = '';
    valueConditionList.forEach((item, index) => {
      if (index === valueConditionList.length - 1) {
        // 最后一个，不把连接符填上
        if (item.rule === 'is None' || item.rule === 'not None') {
          loopcondition += `(${item.v1} ${item.rule}) `;
        } else {
          loopcondition += `(${item.v1} ${item.rule} ${item.v2}) `;
        }
      } else {
        if (item.rule === 'is None' || item.rule === 'not None') {
          loopcondition += `(${item.v1} ${item.rule}) ${item.connect} `;
        } else {
          loopcondition += `(${item.v1} ${item.rule} ${item.v2}) ${item.connect} `;
        }
      }
    });
  }
  result.output += `${padding}${looptype} ${loopcondition}:\n`;
  return result.output;
};

export default transformLoopStatement;
