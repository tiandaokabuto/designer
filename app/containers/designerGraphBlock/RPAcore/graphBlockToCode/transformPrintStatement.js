import cloneDeep from 'lodash/cloneDeep';

/**
 * (已进行单元测试)
 * @param {*} padding
 * @param {*} dataStructure
 * @param {*} result
 * @param {*} moduleMap
 */
const transformPrintStatement = (padding, dataStructure, result, moduleMap) => {
  result.output += `${padding}`;
  moduleMap.set('sendiRPA.logHandler', 'logger');
  const template_string = dataStructure['properties']['required'][0].value;
  const level_string = dataStructure['properties']['required'][1].value;
  const main_func = dataStructure['properties']['optional'][0].value;
  const params = dataStructure['properties']['optional'][1].value;
  if (params) {
    result.output += `logger.${level_string}("[${level_string.toLocaleUpperCase()}]--" + str(${template_string}).${
      main_func || 'format'
    }(${params}))`;
  } else {
    result.output += `logger.${level_string}("[${level_string.toLocaleUpperCase()}]--" + str(${template_string}))`;
  }
  return [result.output, new Map(moduleMap)];
};

export default transformPrintStatement;
