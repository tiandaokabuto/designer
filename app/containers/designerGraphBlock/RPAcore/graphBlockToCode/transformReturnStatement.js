/**
 * (已进行单元测试)
 * @param {*} padding
 * @param {*} dataStructure
 * @param {*} result
 * @param {*} blockNode
 */
const transformReturnStatement = (
  padding,
  dataStructure,
  result,
  blockNode = {},
  options
) => {
  console.log(dataStructure);
  console.log(blockNode);
  const ignore = dataStructure.ignore ? '# ' : '';
  result.output += `${padding}${ignore}`;
  const returnList =
    (blockNode.properties &&
      blockNode.properties.find(item => item.enName === 'output').value) ||
    [];
  const return_string = dataStructure.properties.required[0].value.map(
    item => item.name || 'None'
  );

  return_string.length = returnList.length;
  for (let i = 0; i < return_string.length; i++) {
    if (!return_string[i]) {
      return_string[i] = 'None';
    }
  }
  console.log(returnList);
  console.log(return_string);
  result.output += `return ${return_string.join(', ')}`;
  return result.output;
};

export default transformReturnStatement;
