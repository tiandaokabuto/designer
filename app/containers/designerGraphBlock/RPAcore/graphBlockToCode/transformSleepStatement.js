const transformSleepStatement = (padding, dataStructure, result, moduleMap) => {
  moduleMap.set('time', 'sleep');
  result.output += `${padding}`;
<<<<<<< HEAD
=======
  console.log(dataStructure);
>>>>>>> 6f9b85961bcc756f86005d94e3b2149fec2fa7f3
  const delay = dataStructure['properties']['required'][0].value;
  result.output += `sleep( ${delay} )`;
};

export default transformSleepStatement;
