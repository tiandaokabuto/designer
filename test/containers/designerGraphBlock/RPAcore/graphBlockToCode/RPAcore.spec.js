import React, { useEffect, useState } from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import transformBreakStatement from '../../../../../app/containers/designerGraphBlock/RPAcore/graphBlockToCode/transformBreakStatement';
import transformConditionalStatement from '../../../../../app/containers/designerGraphBlock/RPAcore/graphBlockToCode/transformConditionalStatement';
import transformContinueStatement from '../../../../../app/containers/designerGraphBlock/RPAcore/graphBlockToCode/transformContinueStatement';
import transformCustomCodeStatement from '../../../../../app/containers/designerGraphBlock/RPAcore/graphBlockToCode/transformCustomCodeStatement';
import transformLoopStatement from '../../../../../app/containers/designerGraphBlock/RPAcore/graphBlockToCode/transformLoopStatement';
import transformPrintStatement from '../../../../../app/containers/designerGraphBlock/RPAcore/graphBlockToCode/transformPrintStatement';
import transformReturnStatement from '../../../../../app/containers/designerGraphBlock/RPAcore/graphBlockToCode/transformReturnStatement';
import transformSleepStatement from '../../../../../app/containers/designerGraphBlock/RPAcore/graphBlockToCode/transformSleepStatement';
import transformVariableDeclar from '../../../../../app/containers/designerGraphBlock/RPAcore/graphBlockToCode/transformVariableDeclar';

Enzyme.configure({ adapter: new Adapter() });

const paddingStart = (length) => '    '.repeat(length);

describe('test RPA', () => {
  it('test transformBreakStatement', () => {
    const Component = function ({ padding, dataStructure, result }) {
      return (
        <div>{transformBreakStatement(padding, dataStructure, result)}</div>
      );
    };

    const Rpa = (
      <div>
        <Component
          padding={paddingStart(2)}
          dataStructure={{}}
          result={{
            output: '',
          }}
        />
      </div>
    );

    expect(renderer.create(Rpa).toJSON()).toMatchSnapshot();
  });
  it('test transformContinueStatement', () => {
    const Component = function ({ padding, dataStructure, result }) {
      return (
        <div>{transformContinueStatement(padding, dataStructure, result)}</div>
      );
    };

    const Rpa = (
      <div>
        <Component
          padding={paddingStart(2)}
          dataStructure={{}}
          result={{
            output: '',
          }}
        />
      </div>
    );

    expect(renderer.create(Rpa).toJSON()).toMatchSnapshot();
  });
  it('test transformCustomCodeStatement', () => {
    const Component = function ({ padding, dataStructure, result }) {
      return (
        <div>
          {transformCustomCodeStatement(padding, dataStructure, result)}
        </div>
      );
    };

    const Rpa = (
      <div>
        <Component
          padding={paddingStart(2)}
          dataStructure={{
            codeValue: '这是一段测试的代码',
          }}
          result={{
            output: '',
          }}
        />
      </div>
    );

    expect(renderer.create(Rpa).toJSON()).toMatchSnapshot();
  });
  it('test transformConditionalStatement', () => {
    const Component = function ({ padding, dataStructure, result }) {
      return (
        <div>
          {transformConditionalStatement(padding, dataStructure, result)}
        </div>
      );
    };

    const Rpa = (
      <div>
        <Component
          padding={paddingStart(2)}
          dataStructure={{
            main: 'condition',
            $$typeof: 4,
            text: '条件分支',
            visibleTemplate: '如果满足{{ifcondition}} 则',
            pkg: 'Control',
            properties: {
              optional: [],
              required: [
                {
                  componentType: 0,
                  default: '',
                  cnName: '条件',
                  valueList: [
                    { v1: 'name', v2: '2', rule: '!=', connect: 'and' },
                    { v1: 'sd', v2: 'er', rule: '>', connect: '' },
                  ],
                  enName: 'ifcondition',
                  valueMapping: [
                    { name: '等于', value: '==' },
                    { name: '不等于', value: '!=' },
                    { name: '大于', value: '>' },
                    { name: '小于', value: '<' },
                    { name: '大于等于', value: '>=' },
                    { name: '小于等于', value: '<=' },
                    { name: '空', value: 'is None' },
                    { name: '非空', value: 'not None' },
                  ],
                  tag: 1,
                  value: '',
                  key: '7',
                  desc: '',
                  forceUpdate: 15,
                  _forceUpdate: 15,
                },
              ],
            },
            ifChildren: [],
            elseChildren: [],
            id: 'node_2',
          }}
          result={{
            output: '',
          }}
        />
      </div>
    );

    expect(renderer.create(Rpa).toJSON()).toMatchSnapshot();
  });
  it('test transformLoopStatement', () => {
    const Component = function ({ padding, dataStructure, result }) {
      return (
        <div>{transformLoopStatement(padding, dataStructure, result)}</div>
      );
    };

    const Rpa = (
      <div>
        <Component
          padding={paddingStart(2)}
          dataStructure={{
            main: 'loop',
            $$typeof: 2,
            text: '循环控制',
            visibleTemplate: '循环: 当满足{{loopcondition}} 时',
            pkg: 'Control',
            properties: {
              optional: [],
              required: [
                {
                  componentType: 1,
                  default: '',
                  cnName: '循环类型',
                  enName: 'looptype',
                  valueMapping: [
                    { name: '遍历数组', value: 'for_list' },
                    { name: '遍历字典', value: 'for_dict' },
                    { name: '计次循环', value: 'for_times' },
                    { name: '条件循环', value: 'for_condition' },
                  ],
                  value: 'for_list',
                  _value: 'for_list',
                },
                {
                  componentType: 0,
                  default: '',
                  cnName: '循环条件',
                  valueList: [],
                  enName: 'loopcondition',
                  valueMapping: [
                    { name: '等于', value: '==' },
                    { name: '不等于', value: '!=' },
                    { name: '大于', value: '>' },
                    { name: '小于', value: '<' },
                    { name: '大于等于', value: '>=' },
                    { name: '小于等于', value: '<=' },
                    { name: '空', value: 'is None' },
                    { name: '非空', value: 'not None' },
                  ],
                  tag: 1,
                  value: '',
                  desc: '',
                  forceUpdate: 12,
                  _forceUpdate: 12,
                  for_list: [
                    {
                      id: 'listKeyword',
                      enName: 'value',
                      cnName: '值',
                      value: 'item',
                      paramType: ['String'],
                    },
                    {
                      id: 'listArray',
                      enName: 'arrayRet',
                      cnName: '数组',
                      value: '[1,2,3]',
                      paramType: ['List'],
                    },
                  ],
                  for_dict: [
                    {
                      id: 'dictKey',
                      enName: 'key',
                      cnName: '键',
                      value: '',
                      paramType: ['String'],
                    },
                    {
                      id: 'dictValue',
                      enName: 'value',
                      cnName: '值',
                      value: '',
                      paramType: ['String'],
                    },
                    {
                      id: 'dictVar',
                      enName: 'dictVar',
                      cnName: '字典',
                      value: '',
                      paramType: ['Dictionary'],
                    },
                  ],
                  for_times: [
                    {
                      id: 'timeIndex',
                      enName: 'index',
                      cnName: '索引名称',
                      value: '',
                      paramType: ['String'],
                    },
                    {
                      id: 'timeStartIndex',
                      enName: 'startIndex',
                      cnName: '初始值',
                      value: '',
                      paramType: ['String'],
                    },
                    {
                      id: 'timeEndIndex',
                      enName: 'endIndex',
                      cnName: '结束值',
                      value: '',
                      paramType: ['String'],
                    },
                    {
                      id: 'timeStep',
                      enName: 'step',
                      cnName: '每次增加',
                      value: '',
                      paramType: ['String'],
                    },
                  ],
                },
              ],
            },
            children: [],
            id: 'node_3',
          }}
          result={{
            output: '',
          }}
        />
      </div>
    );

    expect(renderer.create(Rpa).toJSON()).toMatchSnapshot();
  });
  it('test transformPrintStatement', () => {
    const Component = function ({ padding, dataStructure, result }) {
      const [output, moduleMap] = transformPrintStatement(
        padding,
        dataStructure,
        result,
        new Map()
      );
      console.log(moduleMap, 'moduleMap');
      return (
        <div>
          {output} - {moduleMap.get('sendiRPA.logHandler')}
        </div>
      );
    };

    const Rpa = (
      <div>
        <Component
          padding={paddingStart(2)}
          dataStructure={{
            visible: '',
            subtype: 8,
            main: 'print',
            $$typeof: 1,
            text: '控制台打印',
            pkg: 'Control',
            properties: {
              optional: [
                {
                  paramType: 0,
                  componentType: 1,
                  default: 'format',
                  cnName: '替换类型',
                  enName: 'replaceType',
                  valueMapping: [
                    { name: 'format', value: 'format' },
                    { name: 'format_map', value: 'format_map' },
                  ],
                  value: '',
                  desc: '选择打印输出的替换类型',
                },
                {
                  paramType: 0,
                  componentType: 0,
                  default: '',
                  cnName: '传参',
                  enName: 'params',
                  valueMapping: [],
                  value: '3',
                  desc: 'format函数的入参',
                },
              ],
              required: [
                {
                  componentType: 0,
                  default: '',
                  cnName: '模版字符串',
                  enName: 'template_string',
                  value: '"name-{}"',
                },
                {
                  paramType: 0,
                  componentType: 1,
                  default: '',
                  cnName: '日志等级',
                  enName: 'logLevel',
                  valueMapping: [
                    { name: 'INFO', value: 'info' },
                    { name: 'WARNING', value: 'warning' },
                    { name: 'ERROR', value: 'error' },
                    { name: 'DEBUG', value: 'debug' },
                  ],
                  value: 'warning',
                  desc: '选择打印输出日志等级',
                },
              ],
            },
            id: 'node_23',
            userDesc: '',
            _userDesc: '',
          }}
          result={{
            output: '',
          }}
        />
      </div>
    );
    // expect(map.has('sendiRPA.logHandler')).toBeTruthy();
    expect(renderer.create(Rpa).toJSON()).toMatchSnapshot();
  });
  it('test transformReturnStatement', () => {
    const Component = function ({ padding, dataStructure, result, blockNode }) {
      return (
        <div>
          {transformReturnStatement(padding, dataStructure, result, blockNode)}
        </div>
      );
    };

    const Rpa = (
      <div>
        <Component
          padding={paddingStart(2)}
          dataStructure={{
            visible: '',
            subtype: 16,
            main: 'return',
            $$typeof: 1,
            text: '流程块返回',
            pkg: 'Control',
            properties: {
              optional: [],
              required: [
                {
                  componentType: 0,
                  default: '',
                  cnName: '返回值表达式',
                  enName: 'return_string',
                  value: [{ name: 'name', value: '哈哈哈' }],
                },
              ],
            },
            id: 'node_12',
            userDesc: '',
            _userDesc: '',
          }}
          result={{
            output: '',
          }}
          blockNode={{
            shape: 'processblock',
            properties: [
              {
                cnName: '标签名称',
                enName: 'label',
                value: '流程块',
                default: '',
              },
              { cnName: '输入参数', enName: 'param', value: [], default: '' },
              {
                cnName: '流程块返回',
                enName: 'output',
                value: [{ name: 'name', value: '哈哈哈' }],
                default: '',
              },
            ],
            variable: [],
            cards: [
              {
                visible: '',
                subtype: 16,
                main: 'return',
                $$typeof: 1,
                text: '流程块返回',
                pkg: 'Control',
                properties: {
                  optional: [],
                  required: [
                    {
                      componentType: 0,
                      default: '',
                      cnName: '返回值表达式',
                      enName: 'return_string',
                      value: [{ name: 'name', value: '哈哈哈' }],
                    },
                  ],
                },
                id: 'node_12',
                userDesc: '',
                _userDesc: '',
              },
            ],
            pythonCode: 'pass\n',
          }}
        />
      </div>
    );

    expect(renderer.create(Rpa).toJSON()).toMatchSnapshot();
  });
  it('test transformSleepStatement', () => {
    const Component = function ({ padding, dataStructure, result }) {
      const [output, moduleMap] = transformSleepStatement(
        padding,
        dataStructure,
        result,
        new Map()
      );
      return (
        <div>
          {output} - {moduleMap.get('time')}
        </div>
      );
    };

    const Rpa = (
      <div>
        <Component
          padding={paddingStart(2)}
          dataStructure={{
            visible: '',
            subtype: 128,
            module: 'time',
            main: 'sleep',
            $$typeof: 1,
            text: '延迟',
            pkg: 'sleep',
            properties: {
              optional: [],
              required: [
                {
                  paramType: 1,
                  componentType: 0,
                  default: '',
                  cnName: '延迟时间',
                  enName: 'delay',
                  value: '2',
                },
              ],
            },
            id: 'node_1',
            userDesc: '',
            _userDesc: '',
          }}
          result={{
            output: '',
          }}
        />
      </div>
    );
    expect(renderer.create(Rpa).toJSON()).toMatchSnapshot();
  });
  it('test transformVariableDeclar', () => {
    const Component = function ({ padding, dataStructure, result }) {
      return (
        <div>{transformVariableDeclar(padding, dataStructure, result)}</div>
      );
    };

    const Rpa = (
      <div>
        <Component
          padding={paddingStart(2)}
          dataStructure={{
            subtype: 3,
            main: 'vardefine',
            $$typeof: 1,
            text: '变量赋值',
            visibleTemplate: '变量名称{{newVar}} 赋值为 {{init}}',
            pkg: 'Control',
            properties: {
              optional: [],
              required: [
                {
                  componentType: 0,
                  default: '',
                  cnName: '变量名称',
                  enName: 'newVar',
                  value: '新变量1',
                  _value: '新变量1',
                },
                {
                  componentType: 0,
                  default: '',
                  cnName: '变量值',
                  enName: 'init',
                  value: '2323',
                  _value: '2323',
                },
              ],
            },
            id: 'node_1',
            userDesc: '',
            _userDesc: '',
          }}
          result={{
            output: '',
          }}
        />
      </div>
    );

    expect(renderer.create(Rpa).toJSON()).toMatchSnapshot();
  });
});
