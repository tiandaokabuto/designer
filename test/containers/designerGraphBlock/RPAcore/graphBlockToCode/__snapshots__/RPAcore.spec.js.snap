// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test RPA test transformBreakStatement 1`] = `
<div>
  <div>
            break
  </div>
</div>
`;

exports[`test RPA test transformConditionalStatement 1`] = `
<div>
  <div>
            if (name != 2) and (sd &gt; er) :

  </div>
</div>
`;

exports[`test RPA test transformContinueStatement 1`] = `
<div>
  <div>
            continue
  </div>
</div>
`;

exports[`test RPA test transformCustomCodeStatement 1`] = `
<div>
  <div>
            这是一段测试的代码
  </div>
</div>
`;

exports[`test RPA test transformLoopStatement 1`] = `
<div>
  <div>
            for item in [1,2,3]:

  </div>
</div>
`;

exports[`test RPA test transformPrintStatement 1`] = `
<div>
  <div>
            logger.warning("[WARNING]--" + str("name-{}").format(3))
     - 
    logger
  </div>
</div>
`;

exports[`test RPA test transformReturnStatement 1`] = `
<div>
  <div>
            return name
  </div>
</div>
`;

exports[`test RPA test transformSleepStatement 1`] = `
<div>
  <div>
            sleep( 2 )
     - 
    sleep
  </div>
</div>
`;

exports[`test RPA test transformVariableDeclar 1`] = `
<div>
  <div>
            新变量1 = 2323
  </div>
</div>
`;
